rust/plugin-lib/src/base_cache.rs

```diff
@@ -331,7 +331,7 @@ impl ChunkCache {
         // The one 'complex edit' we should probably be handling is
         // the replacement of a single range. This could be a new
         // convenience method on `Delta`?
-        if start < self.offset || start >= self.offset + self.contents.len() {
+        if start < self.offset || start > self.offset + self.contents.len() {
```

Comment:
I'm wondering, if it might be helpful to add a short explanation to the comment above? Otherwise, I could imagine that someone in the future might think `"after our chunk" >= self.offset + self.contents.len()` and changes it back accidentally.

Reply:
good point, updated!